packages/cubejs-api-gateway/src/gateway.ts

```diff
@@ -1042,26 +1043,61 @@ class ApiGateway {
     const requestStarted = new Date();
     try {
       if (!query) {
-        throw new UserError(
-          'A user\'s query must contain a body'
-        );
+        throw new UserError('A user\'s query must contain a body');
       }
 
       if (!Array.isArray(query) && !query.query) {
         throw new UserError(
           'A user\'s query must contain at least one query param.'
         );
       }
-
+      
       query = {
         ...query,
-        requestId: context.requestId
+        requestId: context.requestId,
       };
 
-      if (query.resultFilter?.objectTypes && !Array.isArray(query.resultFilter.objectTypes)) {
-        throw new UserError(
-          'A query.resultFilter.objectTypes must be an array of strings'
-        );
+      const limitCheck = shouldAddLimit(query.query);
+
+      if (limitCheck === null) {
+        throw new UserError('Invalid SQL query');
+      }
```

Comment:
Please remove this after throwing error from helper
https://github.com/cube-js/cube.js/pull/6050/files#r1093531356